sklearn/neighbors/graph.py

```diff
@@ -77,7 +79,13 @@ def kneighbors_graph(X, n_neighbors, mode='connectivity', metric='minkowski',
             ).fit(X)
     else:
         _check_params(X, metric, p, metric_params)
-    return X.kneighbors_graph(X._fit_X, n_neighbors, mode=mode)
+    if mode == "connectivity":
+        warnings.warn("The behavior of marking each sample itself as the "
```

Comment:
Also, these warnings should ideally be asserted for, or at least ignored, where they are triggered in tests.